src/tools/pub/cmd/index.md

```diff
@@ -35,44 +35,65 @@ See [Troubleshooting Pub](/tools/pub/troubleshoot).
 
 Pub's commands fall into the following categories:
 
-## App creation and maintenance
+* [Managing apps](#managing-apps)
+* [Running and serving apps](#running-and-serving-apps)
+* [Building
+  and deploying apps and packages](#building-and-deploying-apps-and-packages)
+
+## Managing apps
 
 Pub provides a number of commands that support
 the creation and maintenance of a Dart application.
 
-* The [`pub cache`](/tools/pub/cmd/pub-cache) command works with your system cache.
-  You can add new packages to your cache,
-  or perform a clean reinstall of all packages in your cache.
-
-* The [`pub deps`](/tools/pub/cmd/pub-deps) command lists all of the dependencies
-  used by a package.
-
-* The [`pub downgrade`](/tools/pub/cmd/pub-downgrade) command retrieves the lowest
-  versions of all the packages that are listed as dependencies
-  used by the application. This is useful for testing the lower range
-  of your package's dependencies.
-
-* The [`pub get`](/tools/pub/cmd/pub-get) command retrieves the packages that are
-  listed as the dependencies for the application.
-
-* The [`pub upgrade`](/tools/pub/cmd/pub-upgrade) command retrieves the latest
-  versions of all the packages that are listed as dependencies
-  used by the application.
-
-## Development
+In this group, the most commonly used commands are `pub get` and
+`pub upgrade`, which retrieve or upgrade dependencies used by a project.
+Every time you modify a pubspec file, run `pub get`
+to make sure the dependencies are up to date. Some IDEs
+perform this step automatically on the creation of a package,
+or any modification of the pubspec.
+
+[`pub cache`](/tools/pub/cmd/pub-cache)
+: Manages pub's local package cache. Use this command to add packages
+  to your cache, or to perform a clean reinstall of all packages in
+  your cache.
+
+[`pub deps`](/tools/pub/cmd/pub-deps)
+: Lists all dependencies used by a package.
+
+[`pub downgrade`](/tools/pub/cmd/pub-downgrade)
+: Retrieves the lowest versions of all the packages that are
+  listed as dependencies used by the application. Used for testing
+  the lower range of your package's dependencies.
+
+[`pub get`](/tools/pub/cmd/pub-get)
+: Retrieves the packages that are listed as the dependencies for
+  the application.
+  If a `pubspec.lock` file already exists, fetches the version
+  of each dependency (if possible) as listed in the lock file.
+  Creates or updates the lock file, as needed.
+
+[`pub upgrade`](/tools/pub/cmd/pub-upgrade)
+: Retrieves the latest version of each package listed
+  as dependencies used by the application. If a `pubspec.lock`
+  file exists, ignores the versions listed in the lock file and fetches
+  the newest versions that honor the constraints in the pubspec.
+  Creates or updates the lock file, as needed.
+
+## Running and serving apps
 
 Pub supports development for web-based apps and command-line apps.
 
```

Comment:
Not very descriptive. How about:

Pub has support for running web apps and command-line apps.